homeassistant/components/hassio/update.py

```diff
@@ -130,13 +130,19 @@ def entity_picture(self) -> str | None:
         """Return the icon of the add-on if any."""
         if not self.available:
             return None
-        if self._addon_data[ATTR_ICON]:
+        # The icon may not be available right away after
+        # manually enabling an entity. So we need use .get()
+        # here to avoid a KeyError.
+        if self._addon_data.get(ATTR_ICON):
```

Comment:
It would be a lot nicer if we used the dispatcher to create entities for an addon instead of reloading the integration as we could avoid this race, but that would require a lot of refactoring so I didn't want to make this PR larger